File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using System.Collections;

namespace WindowsFormsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport1 report = new XtraReport1();
            report.CreateDocument(false);

            int addedPagesCount = 0;
            IDictionaryEnumerator items = report.Groups.GetEnumerator();
            while(items.MoveNext()){
                DictionaryEntry item = (DictionaryEntry)items.Current;
                int i = Convert.ToInt32(item.Value) + addedPagesCount;
                if(i % 2 == 0) {
                    XtraReport2 emptyPage = new XtraReport2();
                    emptyPage.CreateDocument(false);
                    addedPagesCount++;
                    report.Pages.Insert(i+1, emptyPage.Pages[0]);
                }
            }


            ReportPrintTool tool = new ReportPrintTool(report);
            tool.ShowPreviewDialog();
            
        }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports System.Collections

Namespace WindowsFormsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim report As New XtraReport1()
			report.CreateDocument(False)

			Dim addedPagesCount As Integer = 0
			Dim items As IDictionaryEnumerator = report.Groups.GetEnumerator()
			Do While items.MoveNext()
				Dim item As DictionaryEntry = CType(items.Current, DictionaryEntry)
				Dim i As Integer = Convert.ToInt32(item.Value) + addedPagesCount
				If i Mod 2 = 0 Then
					Dim emptyPage As New XtraReport2()
					emptyPage.CreateDocument(False)
					addedPagesCount += 1
					report.Pages.Insert(i+1, emptyPage.Pages(0))
				End If
			Loop


			Dim tool As New ReportPrintTool(report)
			tool.ShowPreviewDialog()

		End Sub
	End Class
End Namespace
